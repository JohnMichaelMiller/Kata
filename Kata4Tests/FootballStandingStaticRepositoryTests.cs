using Microsoft.VisualStudio.TestTools.UnitTesting;
using Kata4;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Reporters;

namespace Kata4.Tests
{
    [TestClass()]
    [UseReporter(typeof(DiffReporter))]
    public class FootballStandingStaticRepositoryTests
    {
        [TestMethod()]
        public void FootballStandingStaticRepositoryInstantiates()
        {
            var repo = new FootballStandingStaticRepository();
            Approvals.VerifyAll(repo, "Football Standing");
        }

        [TestMethod()]
        public void FootballStandingStaticRepositoryHasCorrectMinimumGoalDifferential()
        {
            var repo = new FootballStandingStaticRepository();
            var result =
                from w in repo
                where -w.GoalDifferential ==
                    repo.Min(d => -d.GoalDifferential)
                select w;
                         
            Approvals.VerifyAll(result, "Football Standing");
        }
    }
}