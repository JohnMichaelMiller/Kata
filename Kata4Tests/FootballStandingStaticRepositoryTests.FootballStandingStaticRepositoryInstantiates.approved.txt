Football Standing[0] = Rank:1 Team: Arsenal, Goals For 79, Goals Against: 36, Goal Diff: 43
Football Standing[1] = Rank:2 Team: Liverpool, Goals For 67, Goals Against: 30, Goal Diff: 37
Football Standing[2] = Rank:3 Team: Manchester_U, Goals For 87, Goals Against: 45, Goal Diff: 42
Football Standing[3] = Rank:4 Team: Newcastle, Goals For 74, Goals Against: 52, Goal Diff: 22
Football Standing[4] = Rank:5 Team: Leeds, Goals For 53, Goals Against: 37, Goal Diff: 16
